src/server/gateways/postgres/plugin.ts

```diff
@@ -22,81 +22,116 @@ export default class PostgresGateway extends Gateway {
 
       await client.query(TRANSACTIONS.COMMIT);
 
       return response;
     } catch (err) {
-      // To-do: We should be handling this according to the design.
+      /*
+      This format will allow Winston to log the error message as "Error: [error message]".
+      Additionally, Winston will also save the error stack so we can trace the error.
+      */
+      this.logger.log("error", "Error:", err);
+
       await client.query(TRANSACTIONS.ROLLBACK);
-      this.logger.log("error", "Id lookup failed");
+
       throw err;
     } finally {
       client.release();
     }
   }
 
-  public async readByTicker(ticker: string): Promise<any> {
+  public async readByTicker(ticker: string): Promise<object> {
     const client: PoolClient = await this.pool.connect();
 
     try {
-      const queryText: string = `SELECT * FROM trades WHERE ticker = $1`;
+      const queryText: string = "SELECT * FROM trades WHERE ticker = $1";
 
       const response: QueryResult = await client.query(queryText, [ticker]);
 
       await client.query(TRANSACTIONS.COMMIT);
 
       return response;
     } catch (err) {
+      this.logger.log("error", "Error:", err);
+
       await client.query(TRANSACTIONS.ROLLBACK);
-      this.logger.log("error", "Ticker lookup failed");
+
       throw err;
     } finally {
       client.release();
     }
   }
 
-  public async readByCompany(company: string): Promise<any> {
+  public async readByCompany(company: string): Promise<object> {
     const client: PoolClient = await this.pool.connect();
 
     try {
-      const queryText: string = `SELECT * FROM trades WHERE company_name = $1`;
+      const queryText: string = "SELECT * FROM trades WHERE company_name = $1";
 
       const response: QueryResult = await client.query(queryText, [company]);
 
       await client.query(TRANSACTIONS.COMMIT);
 
       return response;
     } catch (err) {
+      this.logger.log("error", "Error:", err);
+
       await client.query(TRANSACTIONS.ROLLBACK);
-      this.logger.log("error", "Company lookup failed");
+
       throw err;
     } finally {
       client.release();
     }
   }
 
-  public async readByDate(date: string): Promise<any> {
+  public async readByDate(date: string): Promise<object> {
     const client: PoolClient = await this.pool.connect();
 
     try {
-      const queryText: string = `SELECT * FROM trades WHERE created_at = $1`;
+      const queryText: string = "SELECT * FROM trades WHERE created_at = $1";
 
       const response: QueryResult = await client.query(queryText, [date]);
 
       await client.query(TRANSACTIONS.COMMIT);
 
       return response;
     } catch (err) {
+      this.logger.log("error", "Error:", err);
+
       await client.query(TRANSACTIONS.ROLLBACK);
-      this.logger.log("error", "Date lookup failed");
+
+      throw err;
+    } finally {
+      client.release();
+    }
+  }
+
+  public async readByRefrenceNumber(refNum: string): Promise<object> {
```

Comment:
**Comment**: Again, please spell-check your commits before publishing.

Reply:
Spelling should be correct now.